msteams-platform/toolkit/configure-message-extension-capability.md

```diff
@@ -0,0 +1,632 @@
+                        "inputType": "text"
+                    },
+                    {
+                        "name": "text",
+                        "title": "Text",
+                        "description": "Text for the card",
+                        "inputType": "textarea"
+                    }
+                ]
+            },
+            {
+                "id": "shareMessage",
+                "context": [
+                    "message"
+                ],
+                "description": "Test command to run action on message context (message sharing)",
+                "title": "Share Message",
+                "type": "action",
+                "parameters": [
+                    {
+                        "name": "includeImage",
+                        "title": "Include Image",
+                        "description": "Include image in Hero Card",
+                        "inputType": "toggle"
+                    }
+                ]
+            },
+            {
+                "id": "searchQuery",
+                "context": [
+                    "compose",
+                    "commandBox"
+               ],
+               "description": "Test command to run query",
+               "title": "Search",
+                "type": "query",
+                "parameters": [
+                    {
+                        "name": "searchQuery",
+                        "title": "Search Query",
+                        "description": "Your search query",
+                        "inputType": "text"
+                    }
+                ]
+            }
+        ],
+        "messageHandlers": [
+            {
+                "type": "link",
+                "value": {
+                    "domains": [
+                        "*.botframework.com"
+                    ]
+                }
+            }
+        ]
+    }
+]
+```
+
+### Bring message extension code to your project
+
+1. You can bring your message extension app code into your project. If you don't have an app code, use the previously created message extension app project and copy the source code into your current project. We suggest you copy them into a `bot/` folder. The following folder structure is an example of how your project's folder structure looks like:
+
+   |--.vscode/
+   |--appPackage/
+  |--env/
+   |--infra/
+   |--public/
+   |--bot/           <!--message extension source code-->
+   |   |--index.ts
+   |   |--config.ts
+   |   |--teamsBot.ts
+   |   |--package.json
+   |   |--tsconfig.json
+   |   |--web.config
+   |   |--.Webappignore
+   |--src/            <!--your current source code-->
+   |   |--app.ts
+   |   |--static/
+   |   |--views/
+   |--package.json
+   |--tsconfig.json
+   |--teamsapp.local.yml
+   |--teamsapp.yml
+
+   You can create a root `package.json` using command `npm init -y`. We recommend you reorganize the folder structure and create a root `package.json` as given in the following folder structure:
+
+   |--.vscode/
```

Comment:
This isn't coming up in review site. Check the markdown code tag

Reply:
done